### ISSUE TITLE
Question: Difficulty locating R program

### ISSUE THREAD

**Issue by AllanKinnaird** [vigou3/emacs-modified-macos#1]
Over several versions, whenever I started Emacs I got the "Sorry, no version of R could be found on your system" message (and I lived with it - it's clearly not fatal). With the last version (Emacs-25.2-modified-2) I found that I could solve the problem by including (setq-default inferior-R-program-name "/usr/local/bin/R") early in Emacs.app/Contents/Resources/site-lisp/site-start.el  . Similar modifications to other init files were ineffective - the R program path needed to be set very early in the program startup sequence. With the new version (Emacs-25.3-modified-1), this insertion no longer seems to have any effect, and I have the error message again. Am I missing something? I'm using OS X 10.11.6 El Capitan.

**Comment by vigou3** [comment]
Detecting installed versions of R works out of the box thanks to the great work of the ESS folks, provided R is installed in the standard (a.k.a CRAN default) location. Is that your setup? I myself use this distribution and as you may guess, I make sure it works correctly without user intervention. Maybe there is something in your .emacs file that make things go astray?

**Comment by AllanKinnaird** [comment]
Dear Professor Goulet

I initially hit this error message completely “out of the box” (standard CRAN download of R, standard dmg download of your EMACS distribution - for which, much thanks!), some versions back. I may have raised a question at the time, but I’m not sure. I also use RStudio, but I don’t think that has any effect on the problem. Maybe I’m wrong. I think the only major addition to my R system has been the installation of the iNZight package from Auckland U, and I think I was seeing the error before I did that.

As I said, I don’t regard it as a real problem - it’s just that this beautiful system is incorrectly telling me something is wrong, due (clearly) to something in my setup. So don’t worry if I don’t find a solution.

There was a fair amount of traffic on Stackoverflow and other sites about the occurrence of the “could not find R” error message. Practically all of these discussions appeared to find a solution by modifying one of the init files in the setup chain. I think I tried every one of these solutions.

I eventually (while using v25.2) found the solution I indicated, modifying site-start.el within the MacOs app package. I concluded (now I see, possibly erroneously) that the apparent cure was due to the ensured setting of the R program path before the initialisation scripts required ESS to verify the presence of an R program installation.

I’ll review all the .emacs (etc) files in the initialisation chain and see if I can spot any anomalies. I may end up trying a complete reinstall of all the related programs, although I think I tried that at least once in the past.

“Plan C” is that my beautiful 17” 2009 MacBook Pro is reaching the end of its useful life, and I expect to replace it in the next few months. I intend to do a clean install of anything I use on the new machine, rather than risk bringing over unwanted effects from the old machine. That may solve more of my problems than this one!

many thanks
Allan Kinnaird
